pageserver/src/remote_storage/storage_sync/download.rs

```diff
@@ -80,16 +87,19 @@ pub(super) async fn download_timeline<
                     Ok(remote_timeline) => Cow::Owned(remote_timeline),
                     Err(e) => {
                         error!("Failed to download full timeline index: {:?}", e);
+                        // TODO (current) confusing place, we save remote_disk_consistent_lsn
```

Comment:
@SomeoneToIgnore WDYT?

Reply:
Yes, I don't like it too, might be worth straightening up a bit, but won't put too much effort: this code has to go, since based on `s3 list` and etcd is around the corner.

Reply:
s3 list is orthogonal to etcd introduction as I can see, so this is still relevant

Reply:
Sorry, late night comments are hard sometimes for me.

A bit more of an elaboration:

If I've got your ideas about mitigating `s3 list` right, we would access the timeline contents via reading a certain `timeline_dir/index.json` file (naming is arbitrary), thus "download full timeline index" operation could be omitted entirely, if we store the index in the json file.
Alternatively, there's still a way to go with storing the whole remote index entirely in some file and sync those.

etc enforces the switch from `s3 list` towards either of the sync models, since we're finally able to communicate between everything and also limit the number of timelines the pageserver have to care about (due to assigning).

Also Lsm approach at its current form mainly uploads a single file to S3 (unless the PR had changed again, will need to recheck the final form). In general, the whole Lsm approach was supposed to make low level compression work better, ideally making local fs files synced with S3 directly, without compression.

If I remember correctly, alternative approaches from Konstantin planned to upload to S3 incremental db snapshots(?), which is not that many files either.

----

All above are plans for the relatively near future, so I think very soon we most likely won't have any S3 compression or indices anymore, so I still think this code

> worth straightening up a bit, but won't put too much effort

if worth straightening at all.

Reply:
Got your point, I agree with that so I'll remove the comment and we'll see how it goes with new format. Didnt look into details there yet